src/wire/socket_base.cpp

```diff
@@ -126,12 +126,12 @@ bool SocketManager<B>::RefillReadBuffer() {
 
 template <typename B>
 bool SocketManager<B>::FlushWriteBuffer() {
-  fd_set rset;
+  fd_set wset;
   ssize_t written_bytes = 0;
   wbuf.buf_ptr = 0;
   // still outstanding bytes
-  while ((int)wbuf.buf_size - written_bytes > 0) {
-
+  while ((int)wbuf.buf_size > 0) {
+   written_bytes=0;
```

Comment:
Since we never reset written bytes to zero, if we write less than `wbuf.buf_size` bytes during the write, we will never enter the inner loop again (although we update `wbuf.buf_size` and `wbuf.buf_ptr` on 213) because written_bytes is never reset to zero. So we keep updating `wbuf.buf_size` and `wbuf.buf_ptr` even though no additional data has been written after the first write.

Resetting written_bytes to zero at the beginning of the outer loop ensures that we do not reuse the value from a previous iteration of the outer loop.